### PR TITLE
Member groups add/remove

### LINKED ISSUE #346: Member - groups
This page is already created in our gui.
This page is for showing groups of member. This component for showing group is using component `MemberGroupListComponent`, but we already have component for group-list and according to me this `MemberGroupListComponent` is adding no new functinality and is here for nothing. Check if this is true and if it is delete this useless component and for showing groups use group-list. 
Also implement functionality for add and remove group in this page as it is in old gui.

### PR DESCRIPTION
**Pull request by bodnara**
Resolves #346 

* added add/remove option in member groups
* deleted duplicate grouplist
* updated openAPI

### CODE REVIEW

File: apps/admin-gui/src/app/shared/components/dialogs/add-member-group-dialog/add-member-group-dialog.component.html [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<div class="{{theme}}">
+  <h1 mat-dialog-title>{{'DIALOGS.ADD_MEMBER_GROUP.TITLE' | translate}}</h1>
+  <div mat-dialog-content class="dialog-container">
+    <perun-web-apps-groups-list
+      [groups]="groups"
+      [selection]="selection"
+      [hideColumns]="hideColumns"
+      [disableRouting]="true">
+    </perun-web-apps-groups-list>
```
Comment (xkureck) [inline apps/admin-gui/src/app/shared/components/dialogs/add-member-group-dialog/add-member-group-dialog.component.html:9]: also some mat-spinner and loading would be good before the data is loaded.

File: apps/admin-gui/src/app/vos/pages/member-detail-page/member-groups/member-groups.component.ts [hunk 2]
```diff
@@ -41,11 +48,47 @@ export class MemberGroupsComponent implements OnInit {
...
       this.loading = false;
     }, () => this.loading = false);
   }
 
+  addGroup(){
+    const config = getDefaultDialogConfig();
+    config.width = "850px";
+    config.data = {
+      memberId: this.memberId,
+      theme: "facility-theme"
```
Comment (xkureck) [inline apps/admin-gui/src/app/vos/pages/member-detail-page/member-groups/member-groups.component.ts:62]: the theme is not facility-theme

File: apps/admin-gui/src/app/vos/pages/member-detail-page/member-groups/member-groups.component.ts [hunk 3]
```diff
@@ -41,11 +48,47 @@ export class MemberGroupsComponent implements OnInit {
...
+    });
+  }
+
+  removeGroup(){
+    const config = getDefaultDialogConfig();
+    config.width = "650px";
+    config.data = {
+      memberId: this.memberId,
+      groups: this.selection.selected,
+      theme: "facility-theme"
```
Comment (xkureck) [inline apps/admin-gui/src/app/vos/pages/member-detail-page/member-groups/member-groups.component.ts:80]: not facility-theme

File: apps/admin-gui/src/assets/i18n/en.json [hunk 4]
```diff
@@ -1310,6 +1312,21 @@
...
+      "CANCEL": "Cancel",
+      "SUCCESS": "Groups added"
+    },
+    "REMOVE_MEMBER_GROUP": {
+      "TITLE": "Remove groups",
+      "CANCEL": "Cancel",
+      "REMOVE": "Remove",
+      "DESCRIPTION": "Following groups will be removed.",
+      "ASK": "Do you want to proceed?",
+      "SUCCESS": "Groups removed"
```
Comment (xkureck) [inline apps/admin-gui/src/assets/i18n/en.json:1328]: I do not have very good english, but maybe "Groups were removed" would sound better?

File: apps/admin-gui/src/app/vos/components/member-group-list/member-group-list.component.html [hunk 5]
```diff
@@ -1,59 +0,0 @@
...
-            </mat-checkbox>
-          </td>
-        </ng-container>
-        <ng-container matColumnDef="id">
-          <th *matHeaderCellDef mat-header-cell mat-sort-header>{{'MEMBER_DETAIL.GROUPS.TABLE_GROUP_ID' | translate}}</th>
-          <td *matCellDef="let group" class="static-column-size" mat-cell>{{group.id}}</td>
-        </ng-container>
-        <ng-container matColumnDef="name">
-          <th *matHeaderCellDef mat-header-cell mat-sort-header>{{'MEMBER_DETAIL.GROUPS.TABLE_GROUP_NAME' | translate}}</th>
-          <td *matCellDef="let group" mat-cell>{{group.name}}</td>
```
Comment (xkureck) [inline apps/admin-gui/src/app/vos/components/member-group-list/member-group-list.component.html:30]: these tags are not used now(MEMBER_DETAIL.GROUPS.TABLE_GROUP_NAME), delete them in en.json

File: apps/admin-gui/src/app/vos/pages/member-detail-page/member-groups/member-groups.component.ts [hunk 6]
```diff
@@ -19,15 +24,19 @@ export class MemberGroupsComponent implements OnInit {
...
   loading: boolean;
-  tableId = TABLE_MEMBER_DETAIL_GROUPS;
   filterValue = '';
+
+  tableId = TABLE_MEMBER_DETAIL_GROUPS;
   pageSize: number;
+  selection = new SelectionModel<Group>(true, []);
 
   ngOnInit() {
     this.pageSize = this.tableConfigService.getTablePageSize(this.tableId);
```
Comment (xkureck) [inline apps/admin-gui/src/app/vos/pages/member-detail-page/member-groups/member-groups.component.ts:42]: right on top of the ngOnInit method add 
`this.loading = true;`

File: apps/admin-gui/src/app/vos/pages/member-detail-page/member-groups/member-groups.component.ts [hunk 7]
```diff
@@ -41,11 +50,47 @@ export class MemberGroupsComponent implements OnInit {
   refreshTable() {
     this.loading = true;
     this.groupsService.getMemberGroups(this.memberId).subscribe(groups => {
+      this.selection.clear();
```
Comment (xkureck) [inline apps/admin-gui/src/app/vos/pages/member-detail-page/member-groups/member-groups.component.ts:53]: please also add here 
`this.filterValue = '';`

File: apps/admin-gui/src/app/shared/components/dialogs/add-member-group-dialog/add-member-group-dialog.component.ts [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+  hideColumns = [ 'vo', 'menu' ];
+  filterValue = "";
+
+  ngOnInit(): void {
+    this.theme = this.data.theme;
+    this.loading = true;
+
+    this.memberManager.getMemberById(this.data.memberId).subscribe(member => {
+      this.member = member;
```
Comment (xkureck) [inline apps/admin-gui/src/app/shared/components/dialogs/add-member-group-dialog/add-member-group-dialog.component.ts:44]: If I am not mistaken you dont need the whole object member in whole dialog. Only his memberId, thats what you are using and thats what you need. So this request to the backend is useless.

File: apps/admin-gui/src/app/shared/components/dialogs/add-member-group-dialog/add-member-group-dialog.component.ts [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+      this.groupManager.getAllGroups(this.member.voId).subscribe(groups => {
+        this.groups = groups;
+        this.loading = false;
+      });
```
Comment (xkureck) [inline apps/admin-gui/src/app/shared/components/dialogs/add-member-group-dialog/add-member-group-dialog.component.ts:48]:
```suggestion
      }, error => this.loading = false);
```

File: apps/admin-gui/src/app/shared/components/dialogs/add-member-group-dialog/add-member-group-dialog.component.ts [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+    });
+  }
+
+  onAdd() {
+    const groupIds = this.selection.selected.map(group => group.id);
+
+    this.groupManager.addMember(groupIds, this.member.id).subscribe(() => {
+      this.notificator.showSuccess(this.translate.instant('DIALOGS.ADD_MEMBER_GROUP.SUCCESS'));
+      this.dialogRef.close(true);
+    });
```
Comment (xkureck) [inline apps/admin-gui/src/app/shared/components/dialogs/add-member-group-dialog/add-member-group-dialog.component.ts:58]:
```suggestion
    }, error => this.loading = false);
```

File: apps/admin-gui/src/app/shared/components/dialogs/add-member-group-dialog/add-member-group-dialog.component.ts [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+      this.groupManager.getAllGroups(this.member.voId).subscribe(groups => {
+        this.groups = groups;
+        this.loading = false;
+      });
+    });
+  }
+
+  onAdd() {
+    const groupIds = this.selection.selected.map(group => group.id);
+
```
Comment (xkureck) [inline apps/admin-gui/src/app/shared/components/dialogs/add-member-group-dialog/add-member-group-dialog.component.ts:54]:
```suggestion
this.loading = true;
```

File: apps/admin-gui/src/app/shared/components/dialogs/add-member-group-dialog/add-member-group-dialog.component.ts (hunk 8 above)
Comment (bodnara) [reply apps/admin-gui/src/app/shared/components/dialogs/add-member-group-dialog/add-member-group-dialog.component.ts:44]: I also need member's voId that's why I have this backend request here.

**Pull request merged**